src/golang/cmd/server/server/database.go

```diff
@@ -215,3 +216,32 @@ func CreateWriters(dbConfig *database.DatabaseConfig) (*Writers, error) {
 		WorkflowDagResultWriter: workflowDagResultWriter,
 	}, nil
 }
+
+func GetEngineReaders(readers *Readers) *engine.EngineReaders {
```

Comment:
there seems to be some overlaps between engine readers and server readers. Curious about what's the difference and how should the caller decide which one to use?